src/input_cp2k_dft.F

```diff
@@ -9721,4 +9729,143 @@ SUBROUTINE create_cdft_opt_section(section)
 
    END SUBROUTINE create_cdft_opt_section
 
+   SUBROUTINE create_dft_plus_u_section(section)
+      TYPE(section_type), POINTER                        :: section
+
+      TYPE(keyword_type), POINTER                        :: keyword
+
+      CPASSERT(.NOT. ASSOCIATED(section))
+
+      CALL section_create(section, __LOCATION__, &
+                          name="DFT_PLUS_U", &
+                          description="Define the parameters for the DFT+U correction implemented in the pwdft method", &
+                          n_keywords=4, &
+                          repeats=.FALSE.)
+      NULLIFY (keyword)
+
+      CALL keyword_create(keyword, __LOCATION__, &
+                          name="_SECTION_PARAMETERS_", &
+                          description="Controls the activation of the DFT+U section", &
+                          usage="&DFT_PLUS_U ON", &
+                          default_l_val=.FALSE., &
+                          lone_keyword_l_val=.TRUE.)
+      CALL section_add_keyword(section, keyword)
+      CALL keyword_release(keyword)
+
+      ! SIRIUS_HUBBARD_SUBSPACE_NONE 0
+      ! SIRIUS_HUBBARD_SUBSPACE_FULL_ORTHOGONALIZATION 1
+      ! SIRIUS_HUBBARD_SUBSPACE_NORMALIZE 2
+      ! SIRIUS_HUBBARD_SUBSPACE_ORTHOGONALIZE 3
+
+      CALL keyword_create(keyword, __LOCATION__, &
+                          name="SUBSPACE_METHOD", &
+                          description="Select which subspace method to use for generating the hubbard subspace "// &
+                          "NONE : Use bare atomic wave functions "// &
+                          "NORMALIZE : Normalize the wave functions contributing to the hubbard subspace "// &
+                          "ORTHOGONALIZE : Orthogonalize the wave functions contributing to the hubbard subspace "// &
+                          "FULL_ORTHOGONALIZATION : Use all atomic wave functions to orthogonalize the hubbard subspace ", &
```

Comment:
You can use `enum_desc` for the description of enums, see e.g. https://github.com/cp2k/cp2k/blob/bbd14a5fe6b002419d127c8ec63e3062d934e058/src/input_cp2k_dft.F#L276

Reply:
Not sure what those methods are but we are not speaking about the same thing. Anything related to orthogonalization, etc.... controls how the hubbard subspace is generated but there is no analysis of the subspace. SIRIUS has complete implementation of the various hubbard corrections (including the full theory) but it can not be summarized to one word only as many parameters control the different implementations.

Reply:
my apologies, I misunderstood the intent. Sure it is cleaner that way. Thanks.